src/Resolvers/DataValidationRulesResolver.php

```diff
@@ -34,6 +35,15 @@ public function execute(
     ): array {
         $dataClass = $this->dataConfig->getDataClass($class);
 
+        if ($this->isPropertyMorphable($dataClass)) {
+            /**
+             * @var class-string<PropertyMorphableData> $class
+             */
+            $payload = $path->isRoot() ? $fullPayload : Arr::get($fullPayload, $path->get(), []);
+            $class = $class::morph($payload) ?? $class;
+            $dataClass = $this->dataConfig->getDataClass($class);
```

Comment:
Let's rewrite this to:

```suggestion
            $morphedClass = $class::morph(
                $path->isRoot() ? $fullPayload : Arr::get($fullPayload, $path->get(), [])
            );
            
            $dataClass = $this->dataConfig->getDataClass($morphedClass  ?? $class);
```

Reply:
I've updated this 👍